src/qt/bitcoingui.cpp

```diff
@@ -441,6 +441,42 @@ void BitcoinGUI::createActions()
     }
 #endif // ENABLE_WALLET
 
+    m_chain_selection_action = new QAction(tr("Switch &chain"), this);
+    m_chain_selection_action->setStatusTip(tr("Restart application using a different (test) network"));
+    m_chain_selection_menu = new QMenu(this);
+
+    connect(m_chain_selection_menu, &QMenu::aboutToShow, [this] {
+        m_chain_selection_menu->clear();
+        const std::vector<std::pair<QString, QString>> chains = {{"main", "&Bitcoin"}, {"test", "&Testnet"}, {"regtest", "&Regtest"}, {"signet", "&Signet"}};
+        for (auto chain : chains) {
+            const bool is_current = gArgs.GetChainName() == chain.first.toStdString();
+            QAction* action = m_chain_selection_menu->addAction(chain.second);
+            action->setCheckable(true);
+            action->setChecked(is_current);
+            action->setEnabled(!is_current);
+
+            connect(action, &QAction::triggered, [this, chain] {
+                //: Switch to the mainnet, testnet, signet or regtest chain.
+                QMessageBox::StandardButton btnRetVal = QMessageBox::question(this, tr("Switch chain"),
+                    tr("Client restart required to switch chain.") + "<br><br>" + tr("Client will be shut down. Do you want to proceed?"),
```

Comment:
Translators will much appreciate if both strings belong to the same context:
```suggestion
                    tr("Client restart required to switch chain.\n\nClient will be shut down. Do you want to proceed?"),
```

Reply:
Done